geoh/geoh.py

```diff
@@ -23,7 +28,18 @@ def geohashes(geojson={}, precision=6, start_precision=2):
         _geohashes = geohashes_polygon_intersection(polygon, _geohashes)
         p += 1
 
-    return _geohashes
+    if precision_init >= 9:
+        inside_geohashes_coarse =  geohashes_polygon_within(polygon, _geohashes)
+        inside_geohashes = _generate_inner_geohashes_for_geohashes(inside_geohashes_coarse)
+
+        intersect_geohashes_coarse = list(set(_geohashes) - set(inside_geohashes_coarse))
+        intersect_geohashes = _generate_inner_geohashes_for_geohashes(intersect_geohashes_coarse)
+        intersect_geohashes = geohashes_polygon_intersection(polygon, intersect_geohashes)
+
+        return inside_geohashes + intersect_geohashes
+
+    else:
+        return _geohashes
```

Comment:
We could avoid some of the indentation here by reversing the order of the `if` statement:

```python
    if precision_init < 9:
        return _geohashes

    inside_geohashes_coarse =  geohashes_polygon_within(polygon, _geohashes)
    inside_geohashes = _generate_inner_geohashes_for_geohashes(inside_geohashes_coarse)

    intersect_geohashes_coarse = list(set(_geohashes) - set(inside_geohashes_coarse))
    intersect_geohashes = _generate_inner_geohashes_for_geohashes(intersect_geohashes_coarse)
    intersect_geohashes = geohashes_polygon_intersection(polygon, intersect_geohashes)

    return inside_geohashes + intersect_geohashes
```

This organization may be less intuitive though, up to you.

Reply:
Yep. Changed accordingly and passed unit test.